src/senseflow/binary.rs

```diff
@@ -191,39 +167,84 @@ pub fn parse<T: Read + Seek>(stream: &mut T, _size: usize) -> Result<Vec<SampleI
+        &mut map,
+        tag!(parsed GroupId::Gyroscope,     TagId::Data, "Gyroscope data",     Vec_TimeVector3_f64, |v| format!("{:?}", v), gyro, vec![]),
+    );
+
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Accelerometer, TagId::Unit, "Accelerometer unit", String, |v| v.to_string(), "g".into(), Vec::new()),
+    );
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Gyroscope,     TagId::Unit, "Gyroscope unit",     String, |v| v.to_string(), "deg/s".into(), Vec::new()),
+    );
+
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Gyroscope,     TagId::Orientation, "IMU orientation", String, |v| v.to_string(), imu_orientation.into(), Vec::new()),
+    );
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Accelerometer, TagId::Orientation, "IMU orientation", String, |v| v.to_string(), imu_orientation.into(), Vec::new()),
+    );
+
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Magnetometer,  TagId::Data, "Magnetometer data", Vec_TimeVector3_i64f64, |v| format!("{:?}", v), magn, vec![]),
+    );
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Magnetometer,  TagId::Unit, "Magnetometer unit", String, |v| v.to_string(), "μT".into(), Vec::new()),
+    );
+
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Custom("Angle".into()),        TagId::Data, "Angle data", Vec_TimeVector3_f64, |v| format!("{:?}", v), angl, vec![]),
+    );
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Custom("Angle".into()),        TagId::Unit, "Angle unit", String, |v| v.to_string(), "deg".into(),  Vec::new()),
+    );
+
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Quaternion,   TagId::Data, "Quaternion data",   Vec_TimeQuaternion_f64,  |v| format!("{:?}", v), quat, vec![]),
+    );
+
+    util::insert_tag(
+        &mut map,
+        tag!(parsed GroupId::Default, TagId::Metadata, "Metadata", Json, |v| serde_json::to_string(v).unwrap(), metadata, vec![]),
+    );
+    Ok(vec![SampleInfo {
+        timestamp_ms: first_timestamp as f64,
+        duration_ms: (last_timestamp - first_timestamp) as f64,
+        tag_map: Some(map),
+        ..Default::default()
+    }])
 }
 
 fn checksum<T: Read + Seek>(stream: &mut T, item_size: u64) -> Result<Cursor<Vec<u8>>> {
     let mut buf = vec![0u8; item_size as usize];
     stream.read_exact(&mut buf)?;
-    let sum  = stream.read_u8()?;
+    let sum = stream.read_u8()?;
     let init: u8 = 0;
     let calculated_sum = buf.iter().fold(init, |sum, &x| sum.wrapping_add(x));
 
     if calculated_sum == sum {
         Ok(Cursor::new(buf))
     } else {
-        log::error!("Invalid checksum! {} != {} | {}", calculated_sum, sum, crate::util::to_hex(&buf));
+        log::error!(
+            "Invalid checksum! {} != {} | {}",
+            calculated_sum,
+            sum,
+            crate::util::to_hex(&buf)
+        );
```

Comment:
This was fine in a single line, but it's also fine like that, whatever you prefer